### PR TITLE
feature: Environment.yml file has been created

### PR DESCRIPTION
**Pull request by Rupesh-Darimisetti**
- feature: Add environment.yaml to pydastructure project
- to simply install the environment for the project
- Run  following code
``` bash
conda env create --file environment.yml 
conda activate myenv
```

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: myenv
```
Comment (czgdp1807) [inline environment.yml:1]: A better name here? Like, `pyds-dev`?

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+channels:
+  - defaults
+dependencies:
+  - python=3.6
+  - codecov
+  - pytest-cov
+
```
Comment (czgdp1807) [inline environment.yml:8]:
```suggestion
```

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: myenv
+channels:
+  - defaults
+dependencies:
+  - python=3.6
+  - codecov
+  - pytest-cov
```
Comment (czgdp1807) [inline environment.yml:7]: Add documentation dependencies as well.

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: myenv
+channels:
+  - defaults
```
Comment (czgdp1807) [inline environment.yml:3]:
```suggestion
  - conda-forge
```

File: environment.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+dependencies:
+  - python=3.6
```
Comment (czgdp1807) [inline environment.yml:5]:
```suggestion
  - python=3.8
```

File: environment.yml (hunk 3 above)
Comment (czgdp1807) [reply environment.yml:7]: See, `docs/requirements.txt` inside `pydatastructs` root.

Comment (Rupesh-Darimisetti) [reply environment.yml:7]: conda does not have sphinx_readable_theme package in it what to do

Comment (czgdp1807) [reply environment.yml:8]: I think you have already found the solution. Does that work?

File: environment.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+name: pyds-env
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python=3.9
```
Comment (czgdp1807) [inline environment.yml:6]: `3.8` doesn't work?

Comment (Rupesh-Darimisetti) [reply environment.yml:6]: In python 3.8 version the pytest is not working properly on the test cases present in the project.

**Comment by Rupesh-Darimisetti** [comment]
anything i have to modify in the enviroment.yml file, let me know it

File: pydatastructs/linear_data_structures/tests/benchmarks/test_arrays.py [hunk 7]
```diff
@@ -1,7 +1,8 @@
-import random, timeit, functools, os
+import random, timeit, functools, os, pytest
 from pydatastructs import (OneDimensionalArray, Backend,
     DynamicOneDimensionalArray)
 
+@pytest.mark.xfail
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/tests/benchmarks/test_arrays.py:5]:
```suggestion
```

**Pull request merged**